internal/repositories/attestation/attestation.go

```diff
@@ -35,6 +37,9 @@ func New(indexService indexRepoService, chainID uint64, vehicleAddress common.Ad
 
 // GetAttestations fetches attestations for the given vehicle.
 func (r *Repository) GetAttestations(ctx context.Context, vehicleTokenID int, filter *model.AttestationFilter) ([]*model.Attestation, error) {
+	if !auth.ValidAttestationClaim(ctx, filter) {
+		return nil, errorhandler.NewInternalErrorWithMsg(ctx, jwtmiddleware.ErrJWTInvalid, "invalid claims")
```

Comment:
I think my one serious suggestion here is: I wouldn't call this internal. In REST this could be a 4xx, yeah? Specifically, a 403 if it fails your filter-against-claim check. [I do think it would be internal if you got an error on [this line](https://github.com/DIMO-Network/telemetry-api/pull/158/files#diff-4fd36847d3261a6fcbb4d6c4f6b483e89db2c616b6bf45719d34c3fdc665b1fbR58), I mean that's a total breakdown]

Looking at `errorhandler` I guess it is trying to bring in HTTP codes. Everything else looks ace.

Reply:
Sorry, I missed this somehow. I think that's fine!

Reply:
I just completely did not see this function. Great.